vald/evm/rpc/rpcClient.go

```diff
@@ -106,19 +106,21 @@ func NewClient(url string) (Client, error) {
 		return nil, sdkerrors.Wrapf(err, "cannot query latest block number with the given EVM JSON-RPC url %s", url)
 	}
 
-	moonbeamClient := MoonbeamClientImpl{
+	moonbeamClient := &MoonbeamClientImpl{
 		Client: evmClient,
 		url:    url,
 	}
 	if _, err := moonbeamClient.ChainGetFinalizedHead(context.Background()); err == nil {
 		return moonbeamClient, nil
 	}
 
-	eth2Client := Eth2ClientImpl{
+	eth2Client := &Eth2ClientImpl{
```

Comment:
Should we just call this `EthClient` now? :p

Reply:
I'm also puzzled about how to name these things tbh.